service_management/test/repositories/analysis/dao/AnalysisDaoSpec.scala

```diff
@@ -61,52 +28,56 @@ class AnalysisDaoSpec
     dao.insert(a).futureValue
   }
 
-  "AnalysisEventDao" when {
+  "AnalysisDao" when {
 
     "inserting analysis events" should {
       "return the EventId allocated to a single analysis" in {
-        val res = Some(
-          GenericResult(
-            registeredBy = Some(dummyActorId),
-            registeredDate = Some(dateTimeNow),
-            extRef = Some(Seq("foo", "bar", "fizz")),
-            comment = Some("This is a result comment")
-          )
-        )
-
-        saveAnalysis(Some(oid1), res) mustBe MusitSuccess(EventId(1))
+        val gr = Some(dummyGenericResult())
+        saveAnalysis(Some(oid1), gr) mustBe MusitSuccess(EventId(1))
       }
 
-      "return the EventId allocated for an analysis collection" in {
-        val now = Some(dateTimeNow)
-
+      "return the EventId allocated for an analysis collection with many analyses" in {
+        val gr = Some(dummyGenericResult())
         val e1 = dummyAnalysis(Some(oid1))
         val e2 = dummyAnalysis(Some(oid2))
         val e3 = dummyAnalysis(Some(oid3))
-
-        val ac = AnalysisCollection(
-          id = None,
-          analysisTypeId = dummyAnalysisTypeId,
-          eventDate = now,
-          registeredBy = Some(dummyActorId),
-          registeredDate = now,
-          events = Seq(e1, e2, e3)
-        )
+        val ac = dummyAnalysisCollection(gr, e1, e2, e3)
 
         val res = dao.insertCol(ac).futureValue
 
         res.successValue mustBe EventId(2)
       }
+
+      "return the EventId allocated for an analysis collection with one analysis" in {
+        val e1 = dummyAnalysis(Some(oid1))
+        val ac = dummyAnalysisCollection(None, e1)
+
+        val res = dao.insertCol(ac).futureValue
+
+        res.successValue mustBe EventId(6)
+      }
     }
 
     "fetching analysis events" should {
 
-      "return an analysis collection with children" in {
+      "return an analysis collection with many children" in {
         val res = dao.findById(EventId(2)).futureValue
 
         res.successValue.value mustBe an[AnalysisCollection]
         val ac = res.successValue.value.asInstanceOf[AnalysisCollection]
         ac.events must not be empty
+        ac.events.size mustBe 3
```

Comment:
`ac.events must have size(3)` and remove the empty assert
goes for all the asserts with that combo.